### DISCUSSION TITLE
Using TS 5.0 decorators in extensions

### DISCUSSION THREAD

**Discussion by senyai** [microsoft/vscode-discussions#1167] [Extension Development QnA] [answered]
Hello. I've changed decorators to the new ones that TypeScript 5.0 supports. They are strictly typed, and I like them a lot. Everything works when I compile with `tsc`. Unfortunately, `esbuild` lacks support for these new decorators. I tried the `esbuild-plugin-tsc` plugin, but it's giving me more errors. I also tried `webpack`, but I found it too complicated. Has anyone managed to use these new decorators with a bundler?

**Comment by isidorn** [comment]
I do not know. Maybe @mjbvz has ideas.

**Comment by mjbvz** [comment, marked as answer]
Probably a better question for Stack overflow / esbuild  as this isn't specific to VS Code
